app/agent/activities/workspace/clean-up-workspace-instance.ts

```diff
@@ -0,0 +1,46 @@
+import { withActivityHeartbeat } from "../utils";
+
+import type { CreateActivity } from "~/agent/activities/types";
+import { Token } from "~/token";
+
+export type CleanUpWorkspaceInstanceLocalActivity = (args: {
+  workspaceId: bigint;
+  /** Used if there is no workspace instance associated with the workspace */
+  vmInstanceId?: string;
+}) => Promise<void>;
+export const cleanUpWorkspaceInstanceLocal: CreateActivity<
```

Comment:
Perhaps use Agent and ControlPlane here?
cleanUpWorkspaceInstanceLocal -> cleanUpWorkspaceInstanceInAgent
cleanUpWorkspaceInstanceDb     -> cleanUpWorkspaceInstanceInControlPlane

Reply:
Both activities are executed in the agent. I'd rather leave it as is.